### ISSUE TITLE
Extra < character breaks parsing: "matt <3 <matt@gmail.com>"

### ISSUE THREAD

**Issue by mattcasey** [nodemailer/addressparser#10]
For the record, this library has been super-reliable for the majority of the past year. I came across a case yesterday that doesn't work, so I thought I'd share. For now, my plan is to remove/hide orphan "<" before giving it to addressparser:
```
var parsed = addressparse("matt <3 <matt@gmail.com>".replace(/(<)([^>]*<)/g, '&lt$2'))
parsed.forEach(raw => raw.name = raw.name.replace(/&lt/g, '<');
```

**Comment by andris9** [comment]
The "<" in this case is unexpected as it is not allowed at this position. Correct way to include it would be to use quotes: `"matt <3" <matt@gmail.com>` instead of `matt <3 <matt@gmail.com>`.

I checked how other mail clients handle this situation and decided to go the same way OSX Mail does – everything between the unexpected < and the actual address is truncated. So in your case `matt <3 <matt@gmail.com>` becomes `matt <matt@gmail.com>`.

Released the update as v1.0.1
